### ISSUE TITLE
Recognize JavaScript identifiers based on closure externs

### ISSUE THREAD

**Issue by jeluard** [cursiveclojure/cursive#777]
In ClojureScript JavaScript identifiers should be validated based on all accessible closure externs files. This would ensure that if cursive recognizes an identifier it will be properly processed by the ClojureScript compiler in advanced mode.
Currently it is not clear to me based on what logic an identifier is declared valid.

This would make people much more confident about what will happen in `advanced` compilation mode.

**Comment by egret85** [comment]
I agree that support for javascript externs is needed.  Currently most external javascript libraries do not resolve in ClojureScript and the ones that do are inconsistent.

**Comment by timothypratley** [comment]
externs seem to resolve correctly now? is this still a valid issue?
